### ISSUE TITLE
Make this repo work. Bounty Reward: $80

### ISSUE THREAD

**Issue by pvinis** [pvinis/expo-bottom-sheet-playground#2]
Look at the goals, run the app, and fix it.

Bounty Reward: $80

**Comment by pvinis** [comment]
/reward 80
<details><summary>How to earn this $80.00 reward?</summary><br/>💪 Comment `/try` below and start working on it!
🙌 Once your PR is open, comment `/claim #2` either in the PR description or within the comments.</details>

**Comment by pvinis** [comment]
Closest hint has been https://x.com/Baconbrix/status/1816896591988494378.

**Comment by AndreiCalazans** [comment]
I think src/app/bottom/_layout.tsx needs to be something like this:
```
import { BottomSheetNavigationOptions } from "@th3rdwave/react-navigation-bottom-sheet"
import { createBottomSheetNavigator } from "@th3rdwave/react-navigation-bottom-sheet"
import { withLayoutContext } from "expo-router"

const { Navigator } = createBottomSheetNavigator()

export const BottomSheetExtendedNavigator = withLayoutContext<
	BottomSheetNavigationOptions,
	typeof Navigator
>(Navigator)

export default function BottomLayout() {
	return (
		<BottomSheetExtendedNavigator>
			<BottomSheetExtendedNavigator.Screen name="index" />
			<BottomSheetExtendedNavigator.Screen name="other" />
		</BottomSheetExtendedNavigator>
	)
}
```

But I don't think index will render as a bottom sheet unless this entire bottom route is a presentationModal

I will need time to test this which I don't currently have.

Thinking about it I would either go 100% presentationalModal and simply use BottomSheet instead of the react-navigation-bottom-sheet so that when the presentationModal render you animate the bottom sheet. Or figure out how Bacon did with actual routes....

**Comment by pvinis** [comment]
Right. The "presentation modal" way would get the height of the screen (almost). I want the bottom-sheet, that I can control the height to be half the screen.

I have tried and tried the ways of @EvanBacon, and I keep failing, so I thought I'll make this repo to solve this. Maybe he'll grab that 80 bucks lol.

**Comment by AndreiCalazans** [comment]
haha yea, honestly further looking at this repo it seems identical to Evan's repo so the only left mismatch is the versions of bottom-sheet libraries you depend on. His repo  works (https://github.com/EvanBacon/expo-router-layouts-example/tree/efcbdc1b8416fda68e6cd4506525409c3273dac7)  so I guess something broke with the latest version of gorhom/bottom-sheet

**Comment by AndreiCalazans** [comment]
Easiest thing you could do would be to just fork or copy that scaffold code and push it here.

**Comment by dannyhw** [comment]
So thats the issue? it just doesn't show up?

**Comment by pvinis** [comment]
Hey danny. Yea basically yes. I just cannot for the life of me figure out how to use the bottom-sheet with expo-router, and I am willing to pay to outsource it 😅.

Right now I can't make it so that when I tap "Open bottom", a bottom sheet that takes up the space of half the screen shows up, and has a stacknav in it.

**Comment by dannyhw** [comment]
well one thing is that you should turn off dynamic sizing when you want a fixed size, I was able to get this progress with that so far
![image](https://github.com/user-attachments/assets/2d318f28-02f2-44d4-94e8-5620c2943184)

```
export default function BottomLayout() {
	return (
		<BottomSheet screenOptions={{ snapPoints: ["50%"], enableDynamicSizing: false }} />
	)
}
```

**Comment by pvinis** [comment]
totally ok with the dynamic thing turned off. I tried with v4 and v5, and I'm ok with whichever, but obviously v5 would be nice. trying this now..

**Comment by pvinis** [comment]
Ah, ok see, here's the thing. The "bottom initial screen" needs to be in the bottom sheet, like the README says. and tapping the "other" should get the other screen in a stack inside that sheet.

**Comment by dannyhw** [comment]
Yeah just sharing that as I believe it may help, will mess around with it some more later

**Comment by dannyhw** [comment]
/try<details><summary>Check if you're the only one trying 👇</summary><br/>You're the first one to try to solve this issue, go for it! 😉</details>

**Comment by dannyhw** [comment]
Think I got it working, but you have to reverse the stacking order of the layouts
![image](https://github.com/user-attachments/assets/42475c04-914e-40d9-aab7-17d61afaca4e)

https://github.com/pvinis/expo-bottom-sheet-playground/pull/3

**Comment by pvinis** [comment]
no restrictions about moving the navs around 👌. gonna try your PR out in a few min! I'm excited.

**Comment by pvinis** [comment]
Solved with #3, bounty sent. Thanks @dannyhw!
